src/main/scala/org/globalforestwatch/layers/Layer.scala

```diff
@@ -282,14 +305,14 @@ trait OptionalILayer extends OptionalLayer with ILayer {
   /**
     * Define how to fetch data for optional Integer rasters
     */
-  def fetchWindow(window: Extent): OptionalITile = {
-
+  def fetchWindow(windowKey: SpatialKey, windowLayout: LayoutDefinition): OptionalITile = {
     new OptionalITile(cropWindow(for {
       source <- source
       raster <- Either
-        .catchNonFatal(source.synchronized {
-          source.read(window).get.tile.band(0)
-        })
+        .catchNonFatal(
+          source.synchronized {
+            LayoutTileSource.spatial(source, windowLayout).read(windowKey).get.band(0)
```

Comment:
Codacy found an issue: [Usage of get on optional type.](https://app.codacy.com/manual/thomas-maschler/gfw_forest_loss_geotrellis/pullRequest?prid=6292950)